[docs] Improve grammar of autocomplete/autofill section (#26798)
mui-org/material-ui: docs/src/pages/components/autocomplete/autocomplete.md

> The browsers have heuristics to help the users fill the form inputs.

Reviewer: ```suggestion
Browsers have heuristics to help the user fill in form inputs,
```

> However, it can harm the UX of the component.

Reviewer: ```suggestion
however this can harm the UX of the component.
```

> By default, the component disables the **autocomplete** feature (remembering what the user has typed for a given field in a previous session) with the `autoComplete="off"` attribute.

Reviewer: ```suggestion
By default, the component disables the `input` **autocomplete** feature (remembering what the user has typed for a given field in a previous session) with the `autoComplete="off"` attribute.
```
